lib/videoroom/metrics_persistor.ex

```diff
@@ -23,27 +24,100 @@ defmodule VideoRoom.MetricsPersistor do
 
   defp do_start(function, options) do
     {scrape_interval, options} = Keyword.pop(options, :scrape_interval, 1)
-    apply(GenServer, function, [__MODULE__, scrape_interval, options])
+    {store_metrics, options} = Keyword.pop(options, :store_metrics, false)
+
+    apply(GenServer, function, [__MODULE__, [scrape_interval, store_metrics], options])
   end
 
   @impl true
-  def init(scrape_interval) do
+  def init([scrape_interval, store_metrics]) do
     scrape_interval_ms =
       Membrane.Time.seconds(scrape_interval)
       |> Membrane.Time.as_milliseconds()
 
     Process.send_after(self(), :scrape, scrape_interval_ms)
 
-    {:ok, %{scrape_interval_ms: scrape_interval_ms}}
+    {:ok,
+     %{
+       store_metrics: store_metrics,
+       scrape_interval_ms: scrape_interval_ms,
+       prev_report: nil,
+       prev_report_ts: 0
+     }}
   end
 
   @impl true
   def handle_info(:scrape, state) do
-    Reporter.scrape(VideoRoomReporter)
-    |> TimescaleDB.store_report()
+    report = Reporter.scrape(VideoRoomReporter)
+
+    Phoenix.PubSub.broadcast!(
+      VideoRoom.PubSub,
+      "metrics",
+      {:metrics, prepare_fe_report(report, state)}
+    )
```

Comment:
To follow up the changes in `StatsChannel`:
```suggestion
    VideoRoomWeb.Endpoint.broadcast!(
      "stats",
      "metrics",
      %{stats: prepare_fe_report(report, state)}
    )
```